src/java.base/share/classes/sun/security/provider/certpath/OCSP.java

```diff
@@ -245,50 +255,25 @@ static OCSPResponse check(List<CertId> certIds, URI responderURI,
                 ("Content-type", "application/ocsp-request");
             con.setRequestProperty
                 ("Content-length", String.valueOf(bytes.length));
-            out = con.getOutputStream();
-            out.write(bytes);
-            out.flush();
-            // Check the response
-            if (debug != null &&
-                con.getResponseCode() != HttpURLConnection.HTTP_OK) {
-                debug.println("Received HTTP error: " + con.getResponseCode()
-                    + " - " + con.getResponseMessage());
-            }
-            in = con.getInputStream();
-            int contentLength = con.getContentLength();
-            if (contentLength == -1) {
-                contentLength = Integer.MAX_VALUE;
-            }
-            response = new byte[contentLength > 2048 ? 2048 : contentLength];
-            int total = 0;
-            while (total < contentLength) {
-                int count = in.read(response, total, response.length - total);
-                if (count < 0)
-                    break;
-
-                total += count;
-                if (total >= response.length && total < contentLength) {
-                    response = Arrays.copyOf(response, total * 2);
-                }
-            }
-            response = Arrays.copyOf(response, total);
-        } finally {
-            if (in != null) {
-                try {
-                    in.close();
-                } catch (IOException ioe) {
-                    throw ioe;
-                }
-            }
-            if (out != null) {
-                try {
-                    out.close();
-                } catch (IOException ioe) {
-                    throw ioe;
-                }
+            try (OutputStream out = con.getOutputStream()) {
+                out.write(bytes);
+                out.flush();
             }
         }
-        return response;
+
+        // Check the response
+        if (debug != null &&
+            con.getResponseCode() != HttpURLConnection.HTTP_OK) {
+            debug.println("Received HTTP error: " + con.getResponseCode()
+                + " - " + con.getResponseMessage());
+        }
+
+        int contentLength = con.getContentLength();
+        if (contentLength == -1) {
+            contentLength = Integer.MAX_VALUE;
+        }
+
```

Comment:
Is there a leak if the URL connection is not closed?

Reply:
From looking at the javadoc for URLConnection, I think the close has to happen on the Input/OutputStreams.  I have the OutputStream inside a try-with-resources block.  But I think I need to do the same with the InputStream so both are closed at the end of the transaction.

Reply:
I general consider to use HttpURLConnection.disconnect() and I/O close.  Per the  HttpURLConnection class specification, "Each HttpURLConnection instance is used to make a single request but the underlying network connection to the HTTP server may be transparently shared by other instances. Calling the close() methods on the InputStream or OutputStream of an HttpURLConnection after a request may free network resources associated with this instance but has no effect on any shared persistent connection. Calling the disconnect() method may close the underlying socket if a persistent connection is otherwise idle at that time.".

Reply:
In this case I don't think there's a shared persistent connection going on.  The URLConnection object's entire lifecycle is within this one method and for one connection only.  But I do see your point here.  Maybe a try-finally with the disconnect inside the finally might be cleaner instead of try-with-resources around each stream.  I'll give that a spin and see what happens.